### PR TITLE
fix(hand tracking): fix event order when using hands and hide hands i…

### LINKED ISSUE #273: Hand tracking resets hand position if tracking lost causing weird visual effects
If the tracking is lost, hands are reset to (0, 0, 0) position.

**To reproduce:** 
- Go to Hands Tracking Demo: https://codesandbox.io/s/gczkp
- Move your hand behind your head, so it loses tracking
- See downwards

**Actual result:**
You see a hand model in idle state in (0,0,0) coordinates (where you stand)

**Expected result:**
Hand model is hidden

My idea to fix it was to find parent element (armature itself) and if its position is strictly (0,0,0) just hide the hand.
But I find this check too loose: maybe the coordinates will be not exacly 0.
Do you have any ideas?

### PR DESCRIPTION
**Pull request by Ledzz**
Fixes #273 

Me a @saitonakamura checked how the events were sent and found some issues with code not cleaning up, not disposing used resources and not hiding hands when they are not tracking. We decided to add OculusHandModel and XRHandMeshModel to this library so the fix is ready for usage as soon as possible.

### CODE REVIEW

File: src/OculusHandModel.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        this,
+        this.controller,
+        undefined,
+        xrInputSource.handedness,
+        xrInputSource.handedness === 'left' ? this.leftModelPath : this.rightModelPath
+      )
+    }
+  }
+
+  private _onDisconnected: EventListener<Event, 'disconnected', Object3D<Event>> = () => {
```
Comment (saitonakamura) [inline src/OculusHandModel.ts:55]: **suggestion:** lets's also add `if (xrInputSource.hand` guard to not to accidently clear it if some non hand-related disconnect event creeps in

File: src/OculusHandModel.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      button.onClear()
+    }
+
+    if (button.isPressed()) {
+      button.whilePressed()
+    }
+  }
+
+  dispose(): void {
+    this.clear()
```
Comment (saitonakamura) [inline src/OculusHandModel.ts:103]: **suggestion:** this code looks similar to what happens in `_onDisconnected`, maybe let's extract it as a private method of a class and reuse it?

File: src/XRHandMeshModel.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        if (XRJoint.visible) {
+          const position = XRJoint.position
+          bone.position.copy(position)
+          bone.quaternion.copy(XRJoint.quaternion)
+          allInvisible = false
+        }
+      }
+    }
+
+    if (allInvisible && this.scene) {
```
Comment (saitonakamura) [inline src/XRHandMeshModel.ts:98]: **suggestion:** we may want to add a comment explaining why we do it?

**Review by saitonakamura — CHANGES_REQUESTED** [review]

**Review by saitonakamura — APPROVED** [review]
awesome work!

**Comment by saitonakamura** [comment]
@Ledzz let's test examples on codesandbox and report here

**Comment by Ledzz** [comment]
@saitonakamura they all seem to work fine, except player demo, it can't load hdr map, so I guess its unrelated to this code

**Pull request merged**